build/cmake/CCache.cmake

```diff
@@ -0,0 +1,22 @@
+#[[
+    This module enables CCache support by inserting a ccache executable as
+    the compiler launcher for C and C++ if there is a ccache executable availbale
+    on the system.
+
+    CCache support will be automatically enabled if it is found on the system.
+    CCache can be forced on or off by setting the MONGO_USE_CCACHE CMake option to
+    ON or OFF.
+]]
+
+# Find and enable ccache for compiling
+find_program(CCACHE_EXECUTABLE ccache)
+if(CCACHE_EXECUTABLE)
+    message(STATUS "Found ccache: ${CCACHE_EXECUTABLE}")
+    option(MONGO_USE_CCACHE "Use CCache when compiling" ON)
```

Comment:
Nit: Is naming options with the project name a common practice?

I grepped the cmake files for `option (` and all of the existing cmake options are named like `ENABLE_*`. I'd suggest renaming this to `ENABLE_CCACHE` or something similar for consistency.

Reply:
The reason I didn't go with `ENABLE_` names is because these options do not affect the build result, whereas the others do. There's no specific convention on qualifying variable names with the project name, but I've always had a habit of doing it.

Reply:
Unless we take this opportunity to establish a proper naming convention for CMake variables in the C Driver (and also the C++ Driver?), I suppose there is freedom of choice here.

My remarks about `ENABLE_*` naming are not motivated by their corresonding behavior, but from the perspective of users configuring CMake variables via tools such as `ccache` or CMake GUI, where (as far as I know) options the user may care to toggle are all grouped together with the `ENABLE_*` prefix.

I believe there is a UX and documentation argument to be made for consistency on this front, but this may be beyond the scope of this particular PR.